crates/net/types/src/dns_node_record.rs

```diff
@@ -27,6 +29,52 @@ pub struct DNSNodeRecord {
     pub id: PeerId,
 }
 
+/// Retry strategy for DNS lookups.
+#[derive(Clone, Debug, Eq, PartialEq, Hash, SerializeDisplay, DeserializeFromStr)]
+pub struct RetryStrategy {
+    /// The amount of milliseconds between attempts.
+    pub interval: Duration,
+    /// The number of attempts to make before failing.
+    pub attempts: usize,
+}
+
+impl fmt::Display for RetryStrategy {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "interval={},attempts={}", self.interval.as_millis(), self.attempts)
+    }
+}
+
+impl FromStr for RetryStrategy {
+    type Err = eyre::Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        // Parse the comma-separated key-value pairs
+        let mut interval = None;
+        let mut attempts = None;
+        for pair in s.split(',') {
+            let mut parts = pair.split('=');
+            let key = parts.next().ok_or(eyre::eyre!("no key found before '=' delim"))?;
+            let value = parts.next().ok_or(eyre::eyre!("no value found after '=' delim"))?;
+
+            match key {
+                "interval" => interval = Some(Duration::from_millis(value.parse()?)),
+                "attempts" => attempts = Some(value.parse()?),
+                _ => {
+                    return Err(eyre::eyre!(
+                        "Invalid key: {}. Valid keys are 'interval' and 'attempts'",
+                        key
+                    ))
+                }
+            }
+        }
```

Comment:
yeah I think with separate flags we can avoid all this parsing code. we also like the cli to be standardized around clap instead of having relatively unique ways of setting cli args